include/seqan/find/find_hamming_simple.h

```diff
@@ -78,11 +78,26 @@ class Pattern<TNeedle, HammingSimple> {
 
     Pattern() : maxDistance(-1), distance(0), matchNFlags(0) {}
 
+#ifdef SEQAN_CXX11_STANDARD
+    template <typename TNeedle2>
+    Pattern(TNeedle2 && ndl,
+            int k = -1,
+            SEQAN_CTOR_DISABLE_IF(IsSameType<typename std::remove_reference<TNeedle2>::type const &, Pattern const &>))
+                :   maxDistance(-1),
+                    distance(0),
+                    matchNFlags(0)
+    {
+        setHost(*this, std::forward<TNeedle2>(ndl), k);
```

Comment:
Actually setting the distance should not be done using the setHost interface.
I think there are interfaces to set the distance. We rather keep a clean setHost interface and set another feature with another function, especially as the current interface does not require the distance parameter for the initialisation of the Pattern.